### PR TITLE
Refactor state validation and add diagrams

### PR DESCRIPTION
**Pull request by torztomasz**
This pull request refactors the state validation code and diagrams.

### CODE REVIEW

**Review by lucadonnoh — APPROVED** [review]

File: packages/config/src/common/ScalingProjectStateValidation.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  | 'ZK Circuits'
+  | 'Prover Architecture'
+  | 'Verification Keys Generation'
+
+type ScalingProjectStateValidationCategory = {
+  title: CategoryTitle
+  description: string
+}
+
+export type ScalingProjectStateValidation = {
```
Comment (antooni) [inline packages/config/src/common/ScalingProjectStateValidation.ts:11]: Use interface instead

File: packages/config/src/layer2s/index.test.ts [hunk 2]
```diff
@@ -484,6 +484,19 @@ describe('layer2s', () => {
       }
     })
   })
+
+  describe('state validation', () => {
+    describe('every description ends with a dot', () => {
+      for (const layer2 of layer2s) {
+        if (!layer2.stateValidation) continue
+
+        expect(layer2.stateValidation?.description.endsWith('.')).toEqual(true)
```
Comment (antooni) [inline packages/config/src/layer2s/index.test.ts:493]: Is there a need for optional operator?

File: packages/config/src/layer3s/index.test.ts [hunk 3]
```diff
@@ -103,4 +103,17 @@ describe('layer3s', () => {
...
     })
   })
+
+  describe('state validation', () => {
+    describe('every description ends with a dot', () => {
+      for (const layer3 of layer3s) {
+        if (!layer3.stateValidation) continue
+
+        expect(layer3.stateValidation?.description.endsWith('.')).toEqual(true)
+        layer3.stateValidation?.categories.forEach((category) => {
```
Comment (antooni) [inline packages/config/src/layer3s/index.test.ts:113]: Is there a need for optional operator?

File: packages/frontend/src/utils/project/getDiagramImage.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { expect } from 'earl'
+import mockFs from 'mock-fs'
+import path from 'path'
+
+import { getDiagramImage } from './getDiagramImage'
+
+describe(getDiagramImage.name, () => {
+  it('returns the correct path if the file exists', () => {
+    mockFs({
```
Comment (antooni) [inline packages/frontend/src/utils/project/getDiagramImage.test.ts:9]: Do we need this mockFs dependency? Can't we just save some temporary file and at the end delete it?

**Review by antooni — CHANGES_REQUESTED** [review]
Please resolve comments

File: packages/config/src/layer2s/index.test.ts (hunk 2 above)
Comment (torztomasz) [reply packages/config/src/layer2s/index.test.ts:493]: yup, we will not specify state validation for every project

File: packages/config/src/layer3s/index.test.ts (hunk 3 above)
Comment (torztomasz) [reply packages/config/src/layer3s/index.test.ts:113]: yup, see above

**Review by antooni — APPROVED** [review]

**Pull request merged**